src/main/java/org/spongepowered/common/mixin/core/block/tiles/MixinTileEntityFurnace.java

```diff
@@ -56,4 +71,51 @@ public DataContainer toContainer() {
         }
         return container;
     }
+
+    /**
+     * A method to automatically fire FurnaceConsumeFuelEvent
+     *
+     * @param ci  A callback info needed to Inject the method
+     */
+    @Inject(method = "update",at = @At(
+            target = "Lnet/minecraft/tileentity/TileEntityFurnace;getItemBurnTime(Lnet/minecraft/item/ItemStack;)I", value ="INVOKE_ASSIGN"))
+    public void callFurnaceConsumeFuelEvent(CallbackInfo ci)
+    {
+        Furnace furnace=(Furnace) (Object) this;
+        ItemStack fuelStack=(ItemStack)(Object)this.furnaceItemStacks[1];
+        ItemStack burnedItem=fuelStack;
+        burnedItem.setQuantity(1);
+        ItemStack remainingFuel=fuelStack;
+        remainingFuel.setQuantity(remainingFuel.getQuantity()-1);//remove one for present the event like already done
+        if (remainingFuel.getQuantity()<=0) {
+            remainingFuel = null;
+        }
+        int itemBurnTime = TileEntityFurnace.getItemBurnTime((net.minecraft.item.ItemStack) fuelStack);
+        if (itemBurnTime>0){
+            final FurnaceConsumeFuelEvent event = SpongeEventFactory.createFurnaceConsumeFuel(Sponge.getGame(), furnace,
+                    (FurnaceData)  getData(FurnaceData.class)
+                    ,burnedItem, remainingFuel,
+                    null, furnace.getInventory(), furnace.getBlock());
+            if (!Sponge.getGame().getEventManager().post(event)) { //if not cancelled
+                this.furnaceBurnTime=this.currentItemBurnTime= TileEntityFurnace.getItemBurnTime((net.minecraft.item.ItemStack) event.getBurnedItem());
+                if(event.getRemainingFuel().isPresent()) {
+                    remainingFuel= event.getRemainingFuel().get();
+                    remainingFuel.setQuantity(remainingFuel.getQuantity()+1); //add one because The mojang class will decrement this
+                }
+                else {
+                    ItemStackBuilder itemStackBuilder= Sponge.getGame().getRegistry().getItemBuilder();
+                    remainingFuel= itemStackBuilder.itemType(ItemTypes.SPONGE).quantity(1).build(); //set  a arbitrary itemStack with quantity=1,
+                                                                                                    // the Mojang class will delete it
+                }
+                this.furnaceItemStacks[1] = (net.minecraft.item.ItemStack) remainingFuel;
+                return;
```

Comment:
Did you mean for this to return from update()? If so, set `@Inject(cancellable=true)` and replace this with `ci.cancel()`.